src/main/java/nextstep/optional/Expression.java

```diff
@@ -14,12 +16,9 @@ private static boolean matchExpression(Expression e, String expression) {
     }
 
     static Expression of(String expression) {
-        for (Expression v : values()) {
-            if (matchExpression(v, expression)) {
-                return v;
-            }
-        }
-
-        throw new IllegalArgumentException(String.format("%s는 사칙연산에 해당하지 않는 표현식입니다.", expression));
+        return Arrays.stream(Expression.values())
+                .filter(expression1 -> matchExpression(expression1, expression))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(String.format("%s는 사칙연산에 해당하지 않는 표현식입니다.", expression)));
```

Comment:
```suggestion
       private static final Map<String, Expression> valuesMap = Arrays.stream(Expression.values())
                                                                    .collect(Collectors.toUnmodifiableMap(value -> value.expression, Function.identity()));
                                                                   
...(생략)...
 static Expression of(String expression) {
        return Optional.ofNullable(valuesMap.get(expression))
                .orElseThrow(() -> new IllegalArgumentException(String.format("%s는 사칙연산에 해당하지 않는 표현식입니다.", expression)));
    }
                                                                                                                                                                                                                                                                                                                        
```
enum 요소가 많지 않다면 상관없지만, 만약 매번 Stream을 여는게 부담 되는 경우에는
이런 방법도 있을 것 같습니다 😊